File: app/src/main/java/com/matchthree/game/model/BoardConfig.kt
package com.matchthree.game.model

/**
 * Board size and gem-color count. Tuned constants per MECHANICS.md;
 * the generator and engine derive everything from these values.
 */
data class BoardConfig(
    val width: Int = DEFAULT_WIDTH,
    val height: Int = DEFAULT_HEIGHT,
    val gemTypeCount: Int = DEFAULT_GEM_TYPES,
) {
    companion object {
        const val DEFAULT_WIDTH = 9
        const val DEFAULT_HEIGHT = 9
        const val DEFAULT_GEM_TYPES = 6
    }
}

File: Engine/Model/BoardConfig.cs
namespace MatchThree.Engine.Model;

public sealed record BoardConfig(
    int Width = 9,
    int Height = 9,
    int GemTypeCount = 6)
{
    public const int DEFAULT_WIDTH = 9;
    public const int DEFAULT_HEIGHT = 9;
    public const int DEFAULT_GEM_TYPES = 6;
}